models/task/task.go

```diff
@@ -500,6 +523,29 @@ func (t *Task) SetState(s string) {
 	t.notifyState(nil)
 }
 
+func (t *Task) SetTags(tags map[string]string, values *values.Values) error {
+	t.Tags = tags
+	if values == nil {
+		return nil
+	}
+	for k, v := range t.Tags {
+		tempk, err := values.Apply(k, nil, "")
```

Comment:
templating the keys looks like it brings a bunch of problems, eg
```
foo: baz
{{`"foo"`}}: bar
```
Which should have priority? How do we enforce it? It looks pretty loose.
My gut instinct would be to template only on the values.

Reply:
This is backported from the `applyTemplateToMap` function, which template keys and values of a map.
According to the [YAML spec](https://yaml.org/spec/1.2/spec.html#id2764044):

> The content of a mapping node is an unordered set of key: value node pairs, with the restriction that each of the keys is unique.

uTask is using `go-yaml` v2 through its dependency `github.com/ghodss/yaml`, which discard all previous occurences of a duplicate key in a mapping node. See https://github.com/go-yaml/yaml/issues/154.

We can either remove the templating on keys, which puts a limitation on the plugin, or let to the template writers the responsibility of handling such cases.

In your specific example, the first couple `foo:baz` is silently discarded.